### PR TITLE
Add On-Ramp Aggregator a11y improvements

### PR DESCRIPTION
**Pull request by wachunei**
**Description**

This PR improves the interaction of On-Ramp components through screen readers

_1. What is the reason for the change?_
Users reported the inability to properly interact with our UI using screen readers 

_2. What is the improvement/solution?_
The solution is to better define the components that should act as group or not. When acting as group, children elements are not reachable.

**Issue**

https://consensys.slack.com/archives/G8RSKCNCD/p1655139704143279

### CODE REVIEW

File: app/components/UI/FiatOnRampAggregator/components/AmountInput.tsx [hunk 1]
```diff
@@ -40,7 +40,12 @@ const AmountInput: React.FC<Props> = ({
   onPress,
   onCurrencyPress,
 }: Props) => (
-  <Box label={label} onPress={onPress} highlighted={highlighted}>
+  <Box
+    label={label}
+    onPress={onPress}
+    highlighted={highlighted}
+    accessible={false}
+  >
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/AmountInput.tsx:48]: This allows for a11y readers to select amount and currency.

Before it acted as a single button, like the crypto selector:
Screen Shot 2022-07-08 at 13 32 27
 
 Now you can select amount or currency:
 Screen Shot 2022-07-08 at 13 32 34
Screen Shot 2022-07-08 at 13 32 39

File: app/components/UI/FiatOnRampAggregator/components/AssetSelectorButton.tsx [hunk 2]
```diff
@@ -37,8 +37,9 @@ const AssetSelectorButton: React.FC<Props> = ({
   icon,
   onPress,
   highlighted,
+  ...props
 }: Props) => (
-  <Box label={label} onPress={onPress} highlighted={highlighted}>
+  <Box label={label} onPress={onPress} highlighted={highlighted} {...props}>
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/AssetSelectorButton.tsx:42]: This allows a11y props to be passed down to the nested `TouchableOpacity`

File: app/components/UI/FiatOnRampAggregator/components/Quote.tsx [hunk 3]
```diff
@@ -125,13 +125,16 @@ const Quote: React.FC<Props> = ({
...
+        accessible={!highlighted}
+        accessibilityLabel={quote.provider?.name}
       >
         <ListItem.Content>
           <ListItem.Body>
             <ListItem.Title>
               <TouchableOpacity
                 onPress={highlighted ? showInfo : undefined}
                 disabled={!highlighted}
+                accessibilityLabel={quote.provider?.name}
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/Quote.tsx:137]: This allows the Quotes box to act as group when not hightlighted/folded:

Screen Shot 2022-07-08 at 13 35 21

But when expanded, children are reachable:
Screen Shot 2022-07-08 at 13 35 09
Screen Shot 2022-07-08 at 13 35 26

**Review by rickycodes — APPROVED** [review]
looks straight forward enough to me :+1:

**Comment by mobularay** [comment]
@wachunei Has QA feature been completed?

**Comment by wachunei** [comment]
Yes but this will also need a spot check in-device with the screen readers. I'll remove `needs-qa` and add spot check. I will also merge this.

**Pull request merged**